packages/grid/x-data-grid-pro/src/hooks/features/columnPinning/useGridColumnPinningPreProcessors.ts

```diff
@@ -1,43 +1,53 @@
 import * as React from 'react';
-import { useTheme } from '@mui/material/styles';
-import { GridPipeProcessor, useGridRegisterPipeProcessor } from '@mui/x-data-grid/internals';
+import {
+  GridPinnedColumnFields,
+  GridPipeProcessor,
+  gridPinnedColumnsSelector,
+  useGridRegisterPipeProcessor,
+  eslintUseValue,
+  gridVisiblePinnedColumnDefinitionsSelector,
+} from '@mui/x-data-grid/internals';
 import { DataGridProProcessedProps } from '../../../models/dataGridProProps';
-import { gridPinnedColumnsSelector } from './gridColumnPinningSelector';
-import { columnPinningStateInitializer } from './useGridColumnPinning';
-import { GridApiPro, GridPrivateApiPro } from '../../../models/gridApiPro';
-import { filterColumns } from '../../../components/DataGridProVirtualScroller';
+import { GridPrivateApiPro } from '../../../models/gridApiPro';
 
 export const useGridColumnPinningPreProcessors = (
   apiRef: React.MutableRefObject<GridPrivateApiPro>,
   props: DataGridProProcessedProps,
 ) => {
-  const { disableColumnPinning, pinnedColumns: pinnedColumnsProp, initialState } = props;
-  const theme = useTheme();
-  let pinnedColumns = gridPinnedColumnsSelector(apiRef.current.state);
-  if (pinnedColumns == null) {
-    // Since the state is not ready yet lets use the initializer to get which
-    // columns should be pinned initially.
-    const initializedState = columnPinningStateInitializer(
-      apiRef.current.state,
-      { disableColumnPinning, pinnedColumns: pinnedColumnsProp, initialState },
-      apiRef,
-    ) as GridApiPro['state'];
-    pinnedColumns = gridPinnedColumnsSelector(initializedState);
+  const { disableColumnPinning } = props;
+
+  let pinnedColumns: GridPinnedColumnFields | null;
+  if (apiRef.current.state.columns) {
+    pinnedColumns = gridPinnedColumnsSelector(apiRef.current.state);
+  } else {
+    pinnedColumns = null;
   }
 
   const prevAllPinnedColumns = React.useRef<string[]>([]);
 
   const reorderPinnedColumns = React.useCallback<GridPipeProcessor<'hydrateColumns'>>(
     (columnsState) => {
+      eslintUseValue(pinnedColumns);
```

Comment:
I'm terribly annoyed with the `react-hooks` rules. They haven't received work in the last years, they don't make use of typescript to avoid useless code (e.g. skip values whose type extends `React.MutableRefObject<any>` from the deps list...), and they don't allow to be turned off for specific symbols rather than just turn off for the whole effect.